optaplanner-core/src/test/java/org/optaplanner/core/api/score/buildin/bendablebigdecimal/BendableBigDecimalScoreHolderTest.java

```diff
@@ -130,4 +130,18 @@ public void rewardPenalize(boolean constraintMatchEnabled) {
         assertEquals(BendableBigDecimalScore.of(new BigDecimal[]{new BigDecimal("-210.0")}, new BigDecimal[]{new BigDecimal("-90.0"), new BigDecimal("310.0")}), scoreHolder.extractScore(0));
     }
 
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
Thanks for adding the unit coverage. In my experience, this way of checking exceptions is sometimes not precise enough. Let's say the exception will inform about "hardWeights" instead of "softWeights". More fine-grained checking can be very easily achieved with the AssertJ library; or by a try-catch block.

Reply:
or JUnit 5 :)

Reply:
Once we are able to use JUnit 5, I 'll go thought the entire code base to replace all "@Test(excepted = ...)" instances